### ISSUE TITLE
Squiggly heredocs linting

### ISSUE THREAD

**Issue by stripedpumpkin** [bbatsov/rubocop#5594]
Assuming:

```yaml
Layout/IndentHeredoc:
  EnforcedStyle: squiggly
```

## Actual behavior
Consider the two following equivalent definitions. The difference is that in one case the closing delimiter is aligned with the opening delimiter while the second has indentation zero.

Arguably, the first one is the correct style.
 
```ruby
class Squiggly
  def poem
    <<~THE_RAVEN
      Once upon a midnight dreary, while I pondered, weak and weary,
      Over many a quaint and curious volume of forgotten lore—
          While I nodded, nearly napping, suddenly there came a tapping,
      As of some one gently rapping, rapping at my chamber door.
      “’Tis some visitor,” I muttered, “tapping at my chamber door—
                  Only this and nothing more.”
    THE_RAVEN
  end

  def same_poem
    <<~THE_RAVEN
      Once upon a midnight dreary, while I pondered, weak and weary,
      Over many a quaint and curious volume of forgotten lore—
          While I nodded, nearly napping, suddenly there came a tapping,
      As of some one gently rapping, rapping at my chamber door.
      “’Tis some visitor,” I muttered, “tapping at my chamber door—
                  Only this and nothing more.”
THE_RAVEN
  end
end

Squiggly.new.poem == Squiggly.new.same_poem
=> true
```

but rubocop seems perfectly happy with both styles

```bash
$ rubocop -a app/models/squiggly.rb 
Inspecting 1 file
Parser::Source::Rewriter is deprecated.
Please update your code to use Parser::Source::TreeRewriter instead
.

1 file inspected, no offenses detected
```

## Expected behavior
Could possibly be to auto-correct the indentation of the closing delimiter in the second method.

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.5.0.2, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
Closing as duplicate of #5406. Please follow that issue for updates. 🙂
